### ISSUE TITLE
user-db and catalogue-db of the socks store app are exiting

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#5637]
@pescerosso commented on [Mon Jul 03 2017](https://github.com/vmware/vic-planning/issues/63)

[container-logs.zip](https://github.com/vmware/vic-planning/files/1120225/container-logs.zip)
I tried to run the socks store app https://github.com/microservices-demo/microservices-demo/blob/master/deploy/docker-swarm/docker-compose.yml
 
But the  user-db and catalogue-db containers are exiting:
![image](https://user-images.githubusercontent.com/6947684/27802564-5e63bbf6-5fd9-11e7-8064-733a3a4a9747.png)

I am using VIC 1.1.1

**Comment by mhagen-vmware** [comment]
@pescerosso Thanks for filing this issue, we will attempt to reproduce this but if you can reproduce it easily, then please attach a log bundle with the container logs.  Thanks!

**Comment by corrieb** [comment]
Seems to be mostly or all related to #2929. I appreciate that the "lost+found" folder is created by the operating system, but we may want to revisit the decision that was made in that issue. As was pointed out in the issue, this is likely not an issue limited to Postgres

**Comment by anchal-agrawal** [comment]
The `socks_catalogue-db_1` container failed because of the volume mount directory being non-empty (what @corrieb mentioned above):
```
x3Q/sZThzRRgAQ==Initializing database
8oP0s5ThzRTgBQ==2017-07-03T17:17:59.764572Z 0 [Warning] Changed limits: max_open_files: 1024 (requested 5000)
KU5ztJThzRTwBQ==2017-07-03T17:17:59.764656Z 0 [Warning] Changed limits: table_open_cache: 431 (requested 2000)
D6+FtJThzRTgCw==2017-07-03T17:17:59.765665Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
yb2FtJThzRTwBg==2017-07-03T17:17:59.774710Z 0 [ERROR] --initialize specified but the data directory has files in it. Aborting.
xN1atZThzRTwAg==2017-07-03T17:17:59.782263Z 0 [ERROR] Aborting
204qtpThzRQQAA==
```
The `socks_user-db_1` container logs show:

```
vgXbnqPhzRRgBA==warning: database is not yet initialized, and "--config" is specified
wwHlnqPhzRRgAw==  the initdb database startup might fail as a result!
NTDlnqPhzRQQAA==
NklyLKXhzRTABA==about to fork child process, waiting until server is ready for connections.
9C9oLaXhzRRAAQ==forked process: 253
udnXLaXhzRTABA==2017-07-03T17:19:10.539+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
NZrVTqXhzRTQCA==2017-07-03T17:19:11.092+0000 I CONTROL  [initandlisten] MongoDB starting : pid=253 port=27017 dbpath=/data/db-users 64-bit host=9ccfc3c320fa
FoW7T6XhzRSgBA==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten] db version v3.4.4
qtN0UKXhzRTgBg==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten] git version: 888390515874a9debd1b6c5d36559ca86b44babd
splQUaXhzRRABg==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.0.1t  3 May 2016
jvsxUqXhzRTABA==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten] allocator: tcmalloc
EYh/U6XhzRRgBA==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten] modules: none
20OBU6XhzRSwBA==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten] build environment:
UVaBU6XhzRTgBA==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten]     distmod: debian81
sWCBU6XhzRTQBA==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten]     distarch: x86_64
22mBU6XhzRQABQ==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten]     target_arch: x86_64
23OBU6XhzRQAGA==2017-07-03T17:19:11.105+0000 I CONTROL  [initandlisten] options: { config: "/etc/mongodb.conf", net: { bindIp: "127.0.0.1", port: 27017, ssl: { mode: "disabled" } }, processManagement: { fork: true, pidFilePath: "/tmp/tmp.B4Ygn2o5wb" }, storage: { dbPath: "/data/db-users", mmapv1: { smallFiles: true } }, systemLog: { destination: "file", logAppend: true, path: "/proc/237/fd/1" } }
QJRaZ6XhzRTQCg==2017-07-03T17:19:11.503+0000 I STORAGE  [initandlisten] exception in initAndListen: 20 Attempted to create a lock file on a read-only directory: /data/db-users, terminating
qd+paKXhzRRgBg==2017-07-03T17:19:11.503+0000 I NETWORK  [initandlisten] shutdown: going to close listening sockets...
aq7faaXhzRTABQ==2017-07-03T17:19:11.503+0000 I NETWORK  [initandlisten] shutdown: going to flush diaglog...
nF3CaqXhzRRABA==2017-07-03T17:19:11.503+0000 I CONTROL  [initandlisten] now exiting
Qdt3a6XhzRRABQ==2017-07-03T17:19:11.503+0000 I CONTROL  [initandlisten] shutting down with code:100
g+5ebKXhzRSgAw==ERROR: child process failed, exited with error number 100
```

Digging deeper.

**Comment by jialin-li** [comment]
The non empty volume should be fixed when by #5777

**Comment by anchal-agrawal** [comment]
TL;DR The [docker-compose file for this application](https://github.com/microservices-demo/microservices-demo/blob/master/deploy/docker-compose/docker-compose.yml) uses a lot of options that are currently unsupported by VIC engine. I'll summarize the failed containers below.

1. user-db
```
  user-db:
    image: weaveworksdemos/user-db
    hostname: user-db
    restart: always
    cap_drop:
      - all
    cap_add:
      - CHOWN
      - SETGID
      - SETUID
    read_only: true
    tmpfs:
      - /tmp:rw,noexec,nosuid
```

`hostname`(https://github.com/vmware/vic/issues/4195), `restart`(https://github.com/vmware/vic/issues/4033), `cap_add`, `cap_drop`, `read_only` and `tmpfs` are unsupported.

2. catalogue-db

```
  catalogue-db:
    image: weaveworksdemos/catalogue-db
    hostname: catalogue-db
    restart: always
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_ALLOW_EMPTY_PASSWORD=true
      - MYSQL_DATABASE=socksdb
```

`hostname` and `restart` are unsupported. This container failed due to https://github.com/vmware/vic/issues/5777.

3. socks_user-sim

```
  user-sim:
    image: weaveworksdemos/load-test
    cap_drop:
      - all
    read_only: true
    hostname: user-simulator
    command: "-d 60 -r 200 -c 2 -h edge-router"
```

`hostname`, `cap_drop` and `read_only` are unsupported. This container failed because it was unable to resolve the `edge-router` host (the `front-end` container), since the `hostname` docker-compose file option isn't supported.

**Comment by anchal-agrawal** [comment]
Closing since the major missing features (`hostname`, `restart` and the empty volume issue) are known. I've created new follow-on issues for the remaining unsupported options: https://github.com/vmware/vic/issues/6046, https://github.com/vmware/vic/issues/6047.
